packages/core/src/fetchers/fetchExerpienceEntry.spec.ts

```diff
@@ -0,0 +1,128 @@
+import { ContentfulClientApi } from 'contentful';
+import { fetchExperienceEntry } from './fetchExperienceEntry';
+import { compositionEntry } from '../test/__fixtures__/composition';
+import { entries } from '../test/__fixtures__/entities';
+import { describe, it, expect, vi, Mock } from 'vitest';
+
+const mockClient = {
+  getEntries: vi.fn(),
+  getAssets: vi.fn(),
+} as unknown as ContentfulClientApi<undefined>;
+
+describe('fetchExperienceEntry', () => {
+  it('should throw and error if client has not been provided', async () => {
+    try {
+      await fetchExperienceEntry({
+        // @ts-expect-error intentionally setting it to undefined
+        client: undefined,
+        experienceTypeId: 'books',
+        locale: 'en-US',
+        identifier: { slug: 'slug' },
+      });
+    } catch (e) {
+      expect((e as Error).message).toBe(
+        'Failed to fetch experience entities. Required "client" parameter was not provided'
+      );
+    }
```

Comment:
[request change]
This test won't function correctly. If the function doesn't throw an error, we don't call `expect` and thus don't make the test as failed. Also `expect` calls should never be conditionally but also have to be called. Please have a look at the docs of `toThrow` ([here](https://jestjs.io/docs/expect#tothrowerror)) and `rejects` ([here](https://jestjs.io/docs/expect#rejects)).

```suggesti
    await expect(async () => {
      await fetchExperienceEntry({
        // @ts-expect-error intentionally setting it to undefined
        client: undefined,
        experienceTypeId: 'books',
        locale: 'en-US',
        identifier: { slug: 'slug' },
      });
    }).rejects.toThrow('Failed to fetch experience entities. Required "client" parameter was not provided');
```

The same goes for the other broken tests in this file.

Reply:
These tests were brought over from the other project, so they would of had the issue there as well. I'll take a look at the suggestion and improve it.